builder/src/prepare_project.py

```diff
@@ -151,7 +291,263 @@ def supports(self, version):
+        COPY . /src
+        WORKDIR /src
+        RUN dotnet restore --packages /packages
+        RUN dotnet publish -c Release -o /published
+
+        FROM {runtime_image}
+        COPY --from=builder /published /app
+        ENV ASPNETCORE_URLS=http://*:${{PORT}}
+        WORKDIR /app
+        ENTRYPOINT [ "dotnet", "{dll_name}.dll" ]
+        """)
+
+    def __init__(self, root, project):
+        """Initializes the instance of SingleProjectApp.
+
+        Args:
+            root: A string with the path to the directory that contains the project.
+            project: A string with the path to the project file.
+        """
+        self.root = root
+        self.project = project
+
+    def generate_dockerfile(self, version_map, output):
+        """Generates the Dockerfile for the app.
+
+        This method will generate a multi-stage Dockerfile that will
+        first build the app (restore + publish) and then will package
+        up the resulting file into a Docker image.
+
+        Args:
+            version_map: A dictionary that maps versions of the
+                         runtime to the base Docker image to use.
+            output: A string with the path where to save the resulting
+                    Dockerfile.
+
+        """
+        minor_version = self._get_project_runtime_version()
+        if minor_version is None:
+            print('No valid .NET Core runtime version found for the app or it is not a ' +
+                  'supported app.')
+            sys.exit(1)
+
+        base_image = get_base_image(version_map, minor_version)
+        if base_image is None:
+            print('The app requires .NET Core runtime version {0} which is not supported at ' +
+                  'this time.').format(minor_version)
+            sys.exit(1)
+
+        project_name = self._get_project_assembly_name()
+        contents = SingleProjectApp.DOCKERFILE_CONTENTS.format(runtime_image=base_image.image,
+                                                               dll_name=project_name)
+        with open(output, 'wt') as out:
+            out.write(contents)
+
+    def _get_project_assembly_name(self):
+        """Returns the name of the main assembly for the project.
+
+        This method will use the file path as the basis for the name
+        of the assembly.
+
+        Returns:
+            A string with the name of the main assbemly for the project.
+        """
+        basename = os.path.basename(self.project)
+        return os.path.splitext(basename)[0]
+
+    def _get_project_runtime_version(self):
+        """This method returns runtime targeted by this project.
+
+        This method will parse the .csproj file and look for the
+        TargetFramework property if present.
+
+        Returns:
+            A string with the target version.
+        """
+        tree = ET.parse(self.project)
+        root = tree.getroot()
+        framework_element = root.find('./PropertyGroup/TargetFramework')
```

Comment:
Do we want to support a project file that uses the `TargetFrameworks` property?

    <PropertyGroup>
      <TargetFrameworks>netcoreapp1.1;netstandard1.6;net45;net40</TargetFrameworks>
    </PropertyGroup>